extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1021,17 +1091,8 @@ export class AmpA4A extends AMP.BaseElement {
 
     this.isCollapsed_ = false;
 
-    // Allow embed to release its resources.
-    if (this.friendlyIframeEmbed_) {
-      this.friendlyIframeEmbed_.destroy();
-      this.friendlyIframeEmbed_ = null;
-    }
-
     // Remove rendering frame, if it exists.
-    if (this.iframe && this.iframe.parentElement) {
-      this.iframe.parentElement.removeChild(this.iframe);
-      this.iframe = null;
-    }
+    this.destroyFrame();
```

Comment:
Regarding the user swiping back, the refresh cycle will stop on its own until the user sees the new creative anyway. So the worst case is that we refresh a number of slots on a page, which the user will not see unless he swipes back to the original page, at most once per refresh-enabled slot.